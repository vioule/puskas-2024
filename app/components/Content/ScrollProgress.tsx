import { motion, useScroll, useSpring } from "framer-motion";

export default function ScrollProgress() {
  const { scrollYProgress } = useScroll();
  const smoothVelocity = useSpring(scrollYProgress, {
    damping: 10,
    stiffness: 300,
  });
  return (
    <motion.div
      id="scroll-progress"
      className="w-[0.8%] bg-gold h-full absolute bottom-0 origin-bottom-left"
      style={{ scaleY: smoothVelocity }}
    ></motion.div>
  );
}
