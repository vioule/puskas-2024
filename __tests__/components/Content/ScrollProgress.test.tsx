import { render, screen } from "@testing-library/react";
import * as module from "framer-motion";
import ScrollProgress from "@/app/components/Content/ScrollProgress";

jest.spyOn(module, "useScroll").mockReturnValue({
  scrollX: module.motionValue(0),
  scrollXProgress: module.motionValue(0),
  scrollY: module.motionValue(0),
  scrollYProgress: module.motionValue(0.75),
});

test("render ScrollProgress unchanged", () => {
  const { container } = render(<ScrollProgress />);
  expect(container).toMatchSnapshot();
});

test("should have correct scale", () => {
  render(<ScrollProgress />);
  expect(screen.getByTestId("scroll-progress")).toHaveStyle(
    "transform: scaleY(0.75) translateZ(0)"
  );
});
