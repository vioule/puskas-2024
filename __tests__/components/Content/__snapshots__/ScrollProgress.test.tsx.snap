// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`render ScrollProgress unchanged 1`] = `
<div>
  <div
    class="w-[0.8%] bg-gold h-full absolute bottom-0 origin-bottom-left"
    data-testid="scroll-progress"
    id="scroll-progress"
    style="transform: scaleY(0.75) translateZ(0);"
  />
</div>
`;
